janus_core/helpers/mlip_calculators.py

```diff
@@ -224,6 +224,36 @@ def choose_calculator(
         kwargs.setdefault("sevennet_config", None)
         calculator = SevenNetCalculator(model=model_path, device=device, **kwargs)
 
+    elif arch == "orb":
+        from orb_models import __version__
+        from orb_models.forcefield.calculator import ORBCalculator
+        from orb_models.forcefield.graph_regressor import GraphRegressor
+        import orb_models.forcefield.pretrained as orb_ff
+
+        if isinstance(model_path, str):
+            match model_path:
+                case "orb-v2":
+                    model = orb_ff.orb_v2()
+                case "orb-mptraj-only-v2":
+                    model = orb_ff.orb_v2_mptraj_only()
+                case "orb-d3-v2":
+                    model = orb_ff.orb_d3_v2()
+                case "orb-d3-xs-v2":
+                    model = orb_ff.orb_d3_xs_v2()
+                case "orb-d3-sm-v2":
+                    model = orb_ff.orb_d3_sm_v2()
+                case _:
+                    raise ValueError(
+                        "Please specify `model_path`, as there is no "
+                        f"default model for {arch}"
+                    )
+        elif isinstance(model_path, GraphRegressor):
```

Comment:
I think match case lets you do type checks like this too, doesn't it? Is there a nice way to combine it with the specific string checks?

Reply:
see comment above to jacob's abstraction